src/blocks/scratch3_looks.js

```diff
@@ -298,6 +298,33 @@ class Scratch3LooksBlocks {
         };
     }
 
+    /**
+     * Create a text bubble and yield on this thread until it's done. Used for "say/think for () secs".
+     * @param {number} args Block arguments
+     * @param {BlockUtility} util Utility object provided by the runtime.
+     * @param {string} type The type of bubble (say/think)
+     * @private
+     */
+    _bubbleForSecs (args, util, type) {
+        if (util.stackTimerNeedsInit()) {
+            this.runtime.emit('SAY', util.target, type, args.MESSAGE);
+
+            const duration = Math.max(0, 1000 * Cast.toNumber(args.SECS));
+
+            util.stackFrame.bubbleId = this._getBubbleState(util.target).usageId;
+            util.startStackTimer(duration);
+            util.yield();
+        } else if (util.stackTimerFinished()) {
+            // Make sure the bubble we're removing is the same bubble we created.
+            // We don't want to cancel a bubble started from another script.
+            if (util.stackFrame.bubbleId === this._getBubbleState(util.target).usageId) {
+                this._updateBubble(util.target, type, '');
+            }
+        } else {
+            util.yield();
+        }
+    }
+
     say (args, util) {
         // @TODO in 2.0 calling say/think resets the right/left bias of the bubble
         let message = args.MESSAGE;
```

Comment:
Blocked on the change that makes say and think bubbles round the same, so that we don't change the rounding behavior